### DISCUSSION TITLE
[Update] About T2I-Adapters-XL

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#2099] [General]
Users have reported that the t2i-adapter-xl does not support `pixel-perfect` preprocessor. This is an announcement that when you use t2i-adapter-xl, the `pixel-perfect` should be unchecked.

### What is “pixel-perfect”?

The lineart preprocessor will convert your image into a lineart. Lets say your image is 768x768 and you want to use SDXL to generate a 1024x1024 image.

* If you check pixel-perfect, the image will be resized to 1024x1024 before computing the lineart, and the resolution of the lineart is 1024x1024.

* If you uncheck pixel-perfect, the image will be resized to `preprocessor resolution` (by default is 512x512, this default number is shared by sd-webui-controlnet, comfyui, and diffusers) before computing the lineart, and the resolution of the lineart is 512x512. The 512x512 lineart will be stretched to a blurry 1024x1024 lineart for SDXL, losing many details.

The “pixel-perfect” was important for controlnet 1.1 users to get accurate linearts without losing details.

Recently users reported that the new t2i-adapter-xl does not support (is not trained with) “pixel-perfect” images. Some examples are here:

### Examples (uncheck “pixel-perfect”, the model works)

![dog512](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/69007ba7-7bd0-42dd-b856-af9ab99133ed)

a brown dog on grass, winter, photo, high quality
Negative prompt: drawing, anime, low quality, distortion
Steps: 30, Sampler: DPM++ 2M SDE Karras, CFG scale: 7, Seed: 12345, Size: 1024x1024, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, RNG: CPU, ControlNet 0: "Module: lineart_realistic, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Crop and Resize, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced", SGM noise multiplier: True, Version: v1.6.0

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/5c72283a-739a-48ad-9608-addf4c62302e)

### Examples (check “pixel-perfect”, the model does not work)

a brown dog on grass, winter, photo, high quality
Negative prompt: drawing, anime, low quality, distortion
Steps: 30, Sampler: DPM++ 2M SDE Karras, CFG scale: 7, Seed: 12345, Size: 1024x1024, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, RNG: CPU, ControlNet 0: "Module: lineart_realistic, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Crop and Resize, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: True, Control Mode: Balanced", SGM noise multiplier: True, Version: v1.6.0

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/bdde27f2-d74c-49d6-9eee-095dc046d31e)

### Tip 1

For t2i-adapter, uncheck pixel-perfect, use 512 as preprocessor resolution, and select balanced control mode.

Actually, this is already the default setting – you do not need to do anything if you just selected the model. However, many users have a habit to always check “pixel-perfect” rightly after selecting the models. This should be noticed when you use t2i-adapters-xl.

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/28f53b03-f8b9-4ccc-8f07-098ac94b496b)

### Tip 2

If things still do not work, try use 384 as `preprocessor resolution`. Some of those models marked with diffusers are trained with very low resolution. Like this:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/fd3f921b-98ce-421f-82ed-7fe929d183cb)

Below is a screenshot from T2I-Adapters-XL official demo:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/4683425f-e174-441e-a80d-8ed476e55e0a)

**Comment by FurkanGozukara** [comment]
thank you so much 

I shared on reddit , twitter , linkedin , civitAI and medium

**Comment by Kansaku** [comment]
After playing a bit with the T2I Lineart adapters. I was able to use the pixel perfect mode if I used a lower control model weight (like 0.4).
This gave me good results with some quick test pictures. Tho my little test is not enough to say it always works with this method. So ymmv with my aproach.

**Comment by xiodf3** [comment]
123
